testsupport/wait/host.go

```diff
@@ -2002,32 +2002,41 @@ func (a *HostAwaitility) WaitForSubSpace(t *testing.T, spaceRequestName, spaceRe
 
 // WaitForSpaceBinding waits until the SpaceBinding with the given MUR and Space names is available with the provided criteria, if any
 func (a *HostAwaitility) WaitForSpaceBinding(t *testing.T, murName, spaceName string, criteria ...SpaceBindingWaitCriterion) (*toolchainv1alpha1.SpaceBinding, error) {
-	var spacebinding *toolchainv1alpha1.SpaceBinding
-	labels := map[string]string{
-		toolchainv1alpha1.SpaceBindingMasterUserRecordLabelKey: murName,
-		toolchainv1alpha1.SpaceBindingSpaceLabelKey:            spaceName,
-	}
+	var spaceBinding *toolchainv1alpha1.SpaceBinding
 
-	err := wait.Poll(a.RetryInterval, 2*a.Timeout, func() (done bool, err error) {
+	err := wait.Poll(a.RetryInterval, 2*a.Timeout, func() (bool, error) {
 		// retrieve the SpaceBinding from the host namespace
-		spaceBindingList := &toolchainv1alpha1.SpaceBindingList{}
-		if err = a.Client.List(context.TODO(), spaceBindingList, client.MatchingLabels(labels), client.InNamespace(a.Namespace)); err != nil {
+		var err error
+		if spaceBinding, err = a.GetSpaceBinding(murName, spaceName); err != nil {
 			return false, err
 		}
-		if len(spaceBindingList.Items) == 0 {
-			return false, nil
-		}
-		if len(spaceBindingList.Items) > 1 {
-			return false, fmt.Errorf("more than 1 binding for MasterUserRecord '%s' to Space '%s'", murName, spaceName)
-		}
-		spacebinding = &spaceBindingList.Items[0]
-		return matchSpaceBindingWaitCriterion(spacebinding, criteria...), nil
+		return matchSpaceBindingWaitCriterion(spaceBinding, criteria...), nil
 	})
 	// no match found, print the diffs
 	if err != nil {
-		a.printSpaceBindingWaitCriterionDiffs(t, spacebinding, criteria...)
+		a.printSpaceBindingWaitCriterionDiffs(t, spaceBinding, criteria...)
 	}
-	return spacebinding, err
+	return spaceBinding, err
+}
+
+// GetSpaceBinding lists all available SpaceBinding with the given MUR and Space names set as labels. If none is found, then it returns nil, nil
+func (a *HostAwaitility) GetSpaceBinding(murName, spaceName string) (*toolchainv1alpha1.SpaceBinding, error) {
+	labels := map[string]string{
+		toolchainv1alpha1.SpaceBindingMasterUserRecordLabelKey: murName,
+		toolchainv1alpha1.SpaceBindingSpaceLabelKey:            spaceName,
+	}
+
+	spaceBindingList := &toolchainv1alpha1.SpaceBindingList{}
+	if err := a.Client.List(context.TODO(), spaceBindingList, client.MatchingLabels(labels), client.InNamespace(a.Namespace)); err != nil {
+		return nil, err
+	}
+	if len(spaceBindingList.Items) == 0 {
+		return nil, nil
```

Comment:
should we return an error here, or in the caller check if returned spaceBinding is `nil` and return a `false` to the `wait.Poll` function?

Reply:
makes sense, also the caller can decide what he wants todo in case of `nil` SpaceBinding.

Thanks for fixing it.